addons/mail/static/src/core/web/mention_list.js

```diff
@@ -0,0 +1,107 @@
+    static template = "mail.MentionList";
+    static components = {
+        NavigableList,
+    };
+    static props = {
+        onSelect: { type: Function },
+        close: { type: Function, optional: true },
+        type: { type: String },
+    };
+    setup() {
+        this.state = useState({
+            searchTerm: "",
+            options: [],
+        });
+        this.orm = useService("orm");
+        this.store = useState(useService("mail.store"));
+        this.suggestionService = useService("mail.suggestion");
+        this.sequential = useSequential();
+        this.ref = useAutofocus({
+            mobile: true,
+        });
+
+        useEffect(
+            () => {
+                if (!this.state.searchTerm) {
+                    this.state.options = [];
+                    return;
+                }
+                this.sequential(async () => {
+                    await this.suggestionService.fetchSuggestions({
+                        delimiter: this.props.type === "partner" ? "@" : "#",
+                        term: this.state.searchTerm,
+                    });
+                    const suggestions = this.suggestionService.searchSuggestions(
+                        {
+                            delimiter: this.props.type === "partner" ? "@" : "#",
+                            term: this.state.searchTerm,
+                        },
+                        undefined,
+                        true
+                    );
+                    this.state.options = [
+                        ...suggestions.mainSuggestions,
+                        ...suggestions.extraSuggestions,
+                    ];
+                });
+            },
+            () => [this.state.searchTerm]
+        );
+    }
+
+    get placeholder() {
+        switch (this.props.type) {
+            case "channel":
+                return _t("Search for a channel...");
+            case "partner":
+                return _t("Search for an user...");
+            default:
+                return _t("Search...");
+        }
+    }
+
+    get navigableListProps() {
+        const props = {
+            anchorRef: this.ref.el,
+            position: "bottom-fit",
+            placeholder: _t("Loading"),
+            onSelect: this.props.onSelect,
+            options: [],
+        };
+        switch (this.props.type) {
+            case "partner":
+                this.state.options.forEach((option) => {
+                    props.options.push({
+                        label: option.name,
+                        partner: option,
+                    });
+                });
+                break;
+            case "channel": {
+                this.state.options.forEach((option) => {
+                    props.options.push({
+                        label: option.name,
+                        channel: option,
+                    });
+                });
+                break;
+            }
+        }
```

Comment:
Using a `switch` for 2 options doesn't look really useful? Could just be 2 `if`.

Reply:
it's OK here I suppose? It makes no difference in the efficiency and I prefer to use a switch because it makes the code more structured and easy to read ( because in the future, if there is anything more need, it can be extended easily as well ).